src/ducks/classifications.js

```diff
@@ -121,10 +123,81 @@ const createClassification = () => {
   };
 };
 
+const saveAllQueuedClassifications = (dispatch) => {
+  const queue = JSON.parse(localStorage.getItem(FAILED_CLASSIFICATION_QUEUE_NAME));
+  if (queue && queue.length !== 0) {
+    console.log('Saving queued classifications:', queue.length);
+    queue.forEach((classificationData) => {
+      apiClient.type('classifications').create(classificationData).save()
+        .then((actualClassification) => {
+          console.log('Saved classification', actualClassification.id);
+
+          try {  //Fix: IE11 doesn't know what to do with Split.classificationCreated()
+            Split.classificationCreated(actualClassification);
+          } catch (err) { console.error('Split.classificationCreated() error: ', err); }
+
+          actualClassification.destroy();
+          const indexInQueue = queue.indexOf(classificationData);
+          queue.splice(indexInQueue, 1);
+          try {
+            localStorage.setItem(FAILED_CLASSIFICATION_QUEUE_NAME, JSON.stringify(queue));
+            console.info('Saved a queued classification, remaining:', queue.length);
+            console.info('ducks/classifications.js submitClassification() success');
+            dispatch({ type: SUBMIT_CLASSIFICATION_SUCCESS });
+          } catch (error) {
+            console.error('Failed to update classification queue:', error);
+          }
+          const { workflow, subjects } = actualClassification.links;
+          dispatch(addAlreadySeen(workflow, subjects));
+
+        })
+        .catch((error) => {
+          console.error('Failed to save a queued classification:', error);
+          console.error('ducks/classifications.js submitClassification() error: ', err);
+          Rollbar && Rollbar.error &&
+          Rollbar.error('ducks/classifications.js submitClassification() error: ', err);
+          alert('ERROR: Could not submit Classification');  //TODO: better messaging
+          dispatch({ type: SUBMIT_CLASSIFICATION_ERROR });
+          switch (error.status) {
+            case 422: {
```

Comment:
Is there any reason we're only triggering on a 422?

I mean, I see sense in this... maybe triggering on a 401 would cause an issue because maybe the user tried submitting when logged in, then logged out, then the re-submission tries to submit the Classification with invalid credentials.

But I just want to be sure we're catering for the pertinent possibilities. What if the user's WiFi drops? I'm not sure if 422 is the result we'll get.

Reply:
I think this code is correct. It deletes the classification if it was received by Panoptes but failed validation. 

A more elegant solution might validate the classification and repair it e.g.  missing required links or missing the `completed_at` timestamp.

Reply:
Thanks for the correction, I was reading the code wrong - I thought a Classification was only _re-queued_ when a 422 is received. Now I understand it's _removed_ from the queue when it's a 422.